src/org/zaproxy/zap/extension/httpsinfo/ExtensionHttpsInfo.java

```diff
@@ -80,8 +82,9 @@ public void unload() {
 		unloaded = true;
 
 		while (!dialogues.isEmpty()) {
-			// When disposed the dialogue is removed from the list
-			dialogues.get(0).dispose();
+			for (HttpsInfoDialog dialog : dialogues) {
```

Comment:
The user can't (or shouldn't) open new dialogues after the unloading, it's the ones already open that triggers the issue. If you prefer to use the for, it can't be a for each, but a normal one (starting the disposing from the last position, or always the first, which ends up being like the while).